libbpf-tools/tcpretrans.h

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Anton Protopopov
+// Copyright (c) 2021 Red Hat, Inc.
+#ifndef __TCPRETRANS_H
+#define __TCPRETRANS_H
+
+#define MAX_ENTRIES 8192
+
+#define RETRANSMIT  1
+#define TLP         2
+
+struct event {
+	__u8 saddr[16];
+	__u8 daddr[16];
+	__u32 af; // AF_INET or AF_INET6
+	__u32 pid;
+	__u16 dport;
+	__u16 sport;
+	__u64 type;
+	int state;
+};
+
+struct ipv4_flow_key {
+	__u32 saddr;
+	__u32 daddr;
+	__u16 dport;
+	__u16 sport;
+};
+
+struct ipv6_flow_key {
+	__u8 saddr[16];
+	__u8 daddr[16];
+	__u16 dport;
+	__u16 sport;
+};
```

Comment:
Consider merge this two struct and adding a type field to distinguish v4/v6.
This way we can save a map in BPF side.

Reply:
I considered this but I would prefer not to.  The existing tcpretrans from BCC prints all ipv4 then ipv6, having two maps makes that a little easier.

Reply:
I would still suggest that using one map and sort the result in userspace.
This can save a map in BPF side for efficiency and merge two print functions into one in userspace.
WDYT ?

Reply:
Are maps really expensive?  We would need to double the ipv6 map's entries, and that would reserve more memory (is memory reserved for map entries?), my math says it will increase total reservation by 458752 bytes.

In any case, I think it's helpful to have the code structure similar to tcpconnect.c and tcpretrans.py.  Unless there is a compelling reason to change the structs, I propose we keep them the same.